apis/capabilities/v1beta1/policy_types.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2020 Red Hat.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"reflect"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	PolicyKind = "Policy"
+
+	// PolicyInvalidConditionType represents that the combination of configuration
+	// in the PolicySpec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	PolicyInvalidConditionType common.ConditionType = "Invalid"
+
+	// PolicyReadyConditionType indicates the policy has been successfully synchronized.
+	// Steady state
+	PolicyReadyConditionType common.ConditionType = "Ready"
+
+	// PolicyFailedConditionType indicates that an error occurred during synchronization.
+	// The operator will retry.
+	PolicyFailedConditionType common.ConditionType = "Failed"
+)
+
+// PolicySchemaSpec defines the desired Policy schema
+type PolicySchemaSpec struct {
+	// Name is the name of the policy schema
+	Name string `json:"name"`
+
+	// Version is the version of the policy schema
+	Version string `json:"version"`
+
+	// Summary is the summary of the policy schema
+	Summary string `json:"summary"`
```

Comment:
Is this field mandatory?

Reply:
Yes, otherwise, the API returns:

```
{"errors":{"schema":["The property '#/' did not contain a required property of 'summary' in schema http://apicast.io/policy-v1/schema#"]}}
```

Reply:
:+1: